### PR TITLE
Apply platform to TOC when splitting model into member level

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs [hunk 1]
```diff
@@ -308,8 +352,10 @@ private FileModel GenerateNewFileModel(FileModel model, PageViewModel newPage, s
             var initialFile = model.FileAndType.File;
             var extension = Path.GetExtension(initialFile);
             var directory = Path.GetDirectoryName(initialFile);
-            var newFileName = PathUtility.ToValidFilePath(key, '-').Replace('`', '-') + extension;
+            var newFileName = PathUtility.ToValidFilePath(key, '-') + extension;
             var filePath = Path.Combine(directory, newFileName).ToNormalizedPath();
+            var originalFilePath = model.OriginalFileAndType.File + "#" + HttpUtility.UrlEncode(key);
+            var originalFileAndType = new FileAndType(model.OriginalFileAndType.BaseDir, originalFilePath, model.OriginalFileAndType.Type, model.OriginalFileAndType.SourceDir, model.OriginalFileAndType.DestinationDir);
```
Comment (ansyral) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: >originalFileAndType [](start = 16, length = 19)

not used?

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs [hunk 2]
```diff
@@ -130,7 +130,11 @@ private ModelWrapper GenerateNonOverloadPage(PageViewModel page, FileModel model
             item.Metadata[SplitReferencePropertyName] = true;
             var newPage = ExtractPageViewModel(page, new List<ItemViewModel> { item });
             var newModel = GenerateNewFileModel(model, newPage, item.Uid);
-            var tree = ConvertToTreeItem(item);
+            var tree = ConvertToTreeItem(item,
+                new Dictionary<string, object>
+                {
+                    [Constants.PropertyName.Platform] = item.Platform,
+                });
```
Comment (ansyral) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: Q: why platform is overwritten metadata?

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 3]
```diff
@@ -209,6 +209,7 @@ public virtual FileModel LoadIntermediateModel(Stream stream)
         protected virtual void UpdateModelContent(FileModel model)
         {
             var apiModel = ApiBuildOutput.FromModel((PageViewModel)model.Content); // Fill in details
+            var serialzied = JsonUtility.Serialize(apiModel, Formatting.Indented);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: >serialzied [](start = 16, length = 10)

not used?

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs [hunk 4]
```diff
@@ -308,8 +352,10 @@ private FileModel GenerateNewFileModel(FileModel model, PageViewModel newPage, s
             var initialFile = model.FileAndType.File;
             var extension = Path.GetExtension(initialFile);
             var directory = Path.GetDirectoryName(initialFile);
-            var newFileName = PathUtility.ToValidFilePath(key, '-').Replace('`', '-') + extension;
```
Comment (superyyrrzz) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: >Replace('`', '-')  [](start = 68, length = 18)

why not replace "`"?

File: test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                var outputRawModelPath = GetRawModelFilePath("toc.yml");
+                Assert.True(File.Exists(outputRawModelPath));
+                var model = JsonUtility.Deserialize<TocItemViewModel>(outputRawModelPath);
+                Assert.NotNull(model);
+                Assert.Equal(1, model.Items.Count);
+                Assert.Equal("Namespace1.Class1%602.html", model.Items[0].TopicHref);
+                Assert.Equal(1, model.Items[0].Items.Count);
+                Assert.Equal("CatLibrary.Cat%602.%23ctor.html", model.Items[0].Items[0].TopicHref);
+                Assert.Equal("Cat", model.Items[0].Items[0].Name);
+                Assert.Equal(new List<string> { "net2", "net46" }, JArray.FromObject(model.Items[0].Items[0].Metadata[Constants.PropertyName.Platform]).Select(s => s.ToString()).ToList());
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs]: > "net2", "net46" [](start = 47, length = 16)

In yml, the platform has only "net46"?

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs (hunk 3 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: oops


---
In reply to: [102859648](https://github.com/dotnet/docfx/pull/1341#discussion_r102859648) [](ancestors = 102859648)

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs (hunk 2 above)
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: oops, no need to

---
In reply to: [102664712](https://github.com/dotnet/docfx/pull/1341#discussion_r102664712) [](ancestors = 102664712)

File: test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs (hunk 5 above)
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs]: CatLibrary.Cat`2.#ctor(`0) has platform "net2"

---
In reply to: [102863453](https://github.com/dotnet/docfx/pull/1341#discussion_r102863453) [](ancestors = 102863453)

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs (hunk 4 above)
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: seems no need to. 

---
In reply to: [102860136](https://github.com/dotnet/docfx/pull/1341#discussion_r102860136) [](ancestors = 102860136)

File: test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs (hunk 5 above)
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.ManagedReference.WithPlugins.Tests/SplitClassPageToMemberLevelTest.cs]: got it, it's a overload item.

---
In reply to: [102865809](https://github.com/dotnet/docfx/pull/1341#discussion_r102865809) [](ancestors = 102865809,102863453)

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs [hunk 6]
```diff
@@ -172,11 +177,24 @@ private ModelWrapper GenerateOverloadPage(PageViewModel page, FileModel model, I
...
+                    [Constants.PropertyName.Platform] = newPrimaryItem.Platform,
                 });
             return new ModelWrapper(newPage, newModel, tree);
         }
 
+        private List<string> MergePlatform(IEnumerable<ItemViewModel> children)
+        {
+            var platforms = children.Where(s => s.Platform != null).SelectMany(s => s.Platform);
+            var set = new SortedSet<string>(platforms);
+            if (set.Count == 0)
```
Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: >set [](start = 16, length = 3)

check the count before put it in sortedSet?

File: plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs (hunk 4 above)
Comment (vicancy) [inline plugins/Microsoft.DocAsCode.Build.MemberLevelManagedReference/SplitClassPageToMemberLevel.cs]: convert path to clean url

---
In reply to: [102865868](https://github.com/dotnet/docfx/pull/1341#discussion_r102865868) [](ancestors = 102865868,102860136)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**